app/controllers/notifications_controller.rb

```diff
@@ -359,19 +359,16 @@ def current_notifications(scope = notifications_for_presentation)
   end
 
   def notifications_for_presentation
-    eager_load_relation = [{subject: :labels}, {repository: {app_installation: {subscription_purchase: :subscription_plan}}}]
-    scope = current_user.notifications.includes(eager_load_relation)
-
-    @search = Search.new(scope: scope, query: params[:q], params: params)
+    @search = Search.initialize_for_saved_search(query: params[:q], user: current_user, params: params)
```

Comment:
I re-used this scope to make a more efficient query, so I abstracted to a new constructor on search :)